wms/models/datasets/sgrid.py

```diff
@@ -457,7 +458,141 @@ def _spatial_data_subset(self, data, spatial_index):
         columns = spatial_index[1, :]
         data_subset = data[rows, columns]
         return data_subset
-
+    
+    def _avg_to_cell_center(self, data_array, avg_dim, vectorstep):
+        """
+        Given a two-dimensional numpy.array, average
+        adjacent row values (avg_dim=1) or adjacent
+        column values (avg_dim=0) to the grid cell
+        center.
+        
+        :param data_array: 2-dimensional data
+        :type data_array: numpy.array
+        :param int avg_dim: integer specify array axis to be averaged
+        :return: averages
+        :rtype: numpy.array
+        
+        """
+        if avg_dim == 0:
+            da = np.transpose(data_array)
+        else:
+            da = data_array
+        da_trim_low = da[:, 1:][::vectorstep, ::vectorstep]
```

Comment:
@ayan-usgs Because we need to `avg_to_cell_center` before subsetting this won't have a better performance, right?  Or am I missing something here?

The PR https://github.com/sgrid/pysgrid/pull/72 removed all full slices from `pysgrid`. That reduced the data download to half when the user does not access the nodes.  (Which is the case here.)